lib/services/versions-service.ts

```diff
@@ -0,0 +1,137 @@
+///<reference path="../.d.ts"/>
+"use strict";
+import * as helpers from "../common/helpers";
+import * as constants from "../constants";
+
+class VersionsService implements IVersionsService {
+	private projectData: IProjectData;
+
+	constructor(private $fs: IFileSystem,
+		private $npmInstallationManager: INpmInstallationManager,
+		private $injector: IInjector,
+		private $staticConfig: Config.IStaticConfig) {
+		this.projectData = this.getProjectData();
+	}
+
+	public getNativescriptCliVersion(): IFuture<IVersionInformation> {
+		return (() => {
+			let currentCliVersion = this.$staticConfig.version;
+			let latestCliVersion = this.$npmInstallationManager.getLatestVersion(constants.NATIVESCRIPT_KEY_NAME).wait();
+
+			return {
+				componentName: constants.NATIVESCRIPT_KEY_NAME,
+				currentVersion: currentCliVersion,
+				latestVersion: latestCliVersion
+			};
+		}).future<IVersionInformation>()();
+	}
+
+	public getNativescriptCoreModulesVersion(): IFuture<IVersionInformation> {
+		return (() => {
+			let latestTnsCoreModulesVersion = this.$npmInstallationManager.getLatestVersion(constants.TNS_CORE_MODULES_NAME).wait();
+			let nativescriptCoreModulesInfo: IVersionInformation = {
+				componentName: constants.TNS_CORE_MODULES_NAME,
+				latestVersion: latestTnsCoreModulesVersion
+			};
+
+			if (this.projectData) {
+				let projectConfig = this.$fs.readJson(this.projectData.projectFilePath).wait();
+				let currentTnsCoreModulesVersion = projectConfig.dependencies[constants.TNS_CORE_MODULES_NAME];
+				nativescriptCoreModulesInfo.currentVersion = currentTnsCoreModulesVersion;
+
+				return nativescriptCoreModulesInfo;
+			} else {
+				return nativescriptCoreModulesInfo;
```

Comment:
if and else return the same object... remove the else and use single return statement outside of the if :)